cache/src/main/scala/coursier/Cache.scala

```diff
@@ -1119,6 +1233,48 @@ object Cache {
     def checkingUpdatesResult(url: String, currentTimeOpt: Option[Long], remoteTimeOpt: Option[Long]): Unit = {}
   }
 
+  object Logger {
+    // adding new methods to this one, not to break bin compat in 2.10 / 2.11
+    abstract class Extended extends Logger {
+      def downloadLength(url: String, totalLength: Long, alreadyDownloaded: Long, watching: Boolean): Unit = {
+        downloadLength(url, totalLength, 0L)
+      }
+
+      def gettingLength(url: String): Unit = {}
+      def gettingLengthResult(url: String, length: Option[Long]): Unit = {}
+    }
+
+    object Extended {
+      def apply(logger: Logger): Extended =
+        logger match {
+          case e: Extended => e
+          case _ =>
+            new Extended {
+              override def foundLocally(url: String, f: File) =
+                logger.foundLocally(url, f)
+
+              override def downloadingArtifact(url: String, file: File) =
+                logger.downloadingArtifact(url, file)
+
+              override def downloadLength(url: String, length: Long) =
+                logger.downloadLength(url, length)
+              override def downloadLength(url: String, length: Long, alreadyDownloaded: Long) =
+                logger.downloadLength(url, length, alreadyDownloaded)
+
+              override def downloadProgress(url: String, downloaded: Long) =
+                logger.downloadProgress(url, downloaded)
+
+              override def downloadedArtifact(url: String, success: Boolean) =
+                logger.downloadedArtifact(url, success)
+              override def checkingUpdates(url: String, currentTimeOpt: Option[Long]) =
+                checkingUpdates(url, currentTimeOpt)
```

Comment:
I think this is an infinite recursion here. Same for `checkingUpdatesResult` below.